src/main/scala/com/moesif/filter/MoesifApiFilter.scala

```diff
@@ -36,10 +40,48 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
   private val moesifCollectorEndpoint = config.moesifCollectorEndpoint
   private val samplingPercentage = config.samplingPercentage
   private val eventModelBuffer = mutable.ArrayBuffer[EventModel]()
+  private val client = new MoesifAPIClient(moesifApplicationId, moesifCollectorEndpoint)
+  private val moesifApi = client.getAPI
+  private var lastTimeAppConfigFetched: DateTime = DateTime.parse("1970-01-01", DateTimeFormat.forPattern("yyyy-MM-dd"))
+  private var isAppConfigFetched: Boolean = false
+  private var appConfigModel = new AppConfigModel
 
   private val logger = Logger.getLogger("moesif.play.filter.MoesifApiFilter")
   logger.info(s"config  is $config")
 
+  def getSampleRateToUse(userId: String, companyId: String): Int = {
+    var sampleRate = appConfigModel.getSampleRate
+    if (userId != null && appConfigModel.getUserSampleRate.containsKey(userId)) sampleRate = appConfigModel.getUserSampleRate.get(userId)
```

Comment:
nitpic: Can be refactored like below. nil checks are unecessary i think
appConfigModel.getUserSampleRate.getOrElse{userId,
appConfigModel.getUserSampleRate.getOrElse(companyId, appConfigModel.getSampleRate)
}